### PR TITLE
[Driver][SYCL] Address issue with improper bundler call with -fsycl-link

### PR DESCRIPTION
**Pull request by mdtoguchi**
The initial change to modify -fsycl-link for AOT introduced a modification to the general toolchain flow that is more aligned with the host link step.  This caused a disconnect with the unbundling step from objects where we were unbundling for host and target, but not providing the needed output files.

Fix this up to allow for both the host and target output files to be available.  There is an additional fix that will be needed to restrict this (i.e. only unbundle the target, and not the host), but functionally what we have in place takes care of the immediate problem at hand.

### CODE REVIEW

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -6749,9 +6749,8 @@ class OffloadingActionBuilder final {
         return false;
       if (HasFPGATarget && !updateInputForFPGA(A, InputArg, Args))
         return false;
-      // FIXME - unbundling action is being split into two different actions
-      // when unbundling objects.  One action for the host, the other for the
-      // device.
+      // FIXME - unbundling action with -fsycl-link is unbunding for both host
```
Comment (srividya-sundaram) [inline clang/lib/Driver/Driver.cpp:6752]: spelling nit: *unbundling

**Review by sarnex — APPROVED** [review]
LGTM, just minor yelling at the clouds

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -5050,7 +5050,7 @@ class OffloadingActionBuilder final {
           for (auto SDA : SYCLDeviceActions)
             SYCLLinkBinaryList.push_back(SDA);
           if (WrapDeviceOnlyBinary)
-            return ABRT_Success;
+            return ABRT_Ignore_Host;
```
[the remark this answers is not in the document]
Comment (mdtoguchi) [reply clang/lib/Driver/Driver.cpp:5053]: This change is mostly for cleaning up what is done during linking later on.  When marked as 'success' we had to gather the items to push to device linking for `-fsycl-link` behaviors where we ended up with 2 different vectors to check.  Now there is only 1 vector making this a little cleaner in implementation.  The core fix is when populating the inputs for the unbundling step.

**Review by hchilama — APPROVED** [review]

**Comment by mdtoguchi** [comment]
@intel/llvm-gatekeepers, this one looks ready to merge, thanks!

**Pull request merged**